internal/server/analytics/clickhouse/client.go

```diff
@@ -0,0 +1,178 @@
+package clickhouse
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/golang-migrate/migrate/v4"
+	clickhouseMigrate "github.com/golang-migrate/migrate/v4/database/clickhouse"
+	"go.uber.org/zap"
+)
+
+// Step defines the value and interval name of the time windows
+// for the Clickhouse query.
+type Step struct {
+	intervalValue int
+	intervalStep  string
+}
+
+var dbOnce sync.Once
+
+const (
+	counterAnalyticsTable = "flipt_counter_analytics"
+	counterAnalyticsName  = "flag_evaluation_count"
+	timeFormat            = "2006-01-02 15:04:05"
+)
+
+type Client struct {
+	conn *sql.DB
+}
+
+// New constructs a new clickhouse client that conforms to the analytics.Client contract.
+func New(logger *zap.Logger, connectionString string) (*Client, error) {
+	var (
+		conn          *sql.DB
+		clickhouseErr error
+	)
+
+	dbOnce.Do(func() {
+		err := runMigrations(logger, connectionString)
+		if err != nil {
+			clickhouseErr = err
+			return
+		}
+
+		connection, err := connect(connectionString)
+		if err != nil {
+			clickhouseErr = err
+			return
+		}
+
+		conn = connection
```

Comment:
We could probably establish the connection once, instead of reconnecting after performing migrations:
```suggestion
		conn, clickhouseErr = connect(connectionString)
		if clickhouseErr != nil {
			return
		}
		
		clickhouseErr = runMigrations(logger, conn)
		return
```

Reply:
@GeorgeMac Got you. I was following the same pattern as the RDMS ones too but it does make sense to just use the connection once as you suggested.